modules/tinymce/src/plugins/link/main/ts/core/NormalizeLink.ts

```diff
@@ -0,0 +1,43 @@
+import DomTreeWalker from 'tinymce/core/api/dom/TreeWalker';
+import Editor from 'tinymce/core/api/Editor';
+
+const isBR = (elm: Node): elm is HTMLAnchorElement => elm && elm.nodeName.toLowerCase() === 'br';
+const isAnchor = (elm: Node): elm is HTMLAnchorElement => elm && elm.nodeName.toLowerCase() === 'a';
+
+const getSelectedLink = (editor: Editor, selectedElm?: Element): HTMLAnchorElement | null => {
+  const isShortEnded = (elm: Node) => elm && editor.schema.getShortEndedElements().hasOwnProperty(elm.nodeName);
+  const isSameSiblingAnchor = (elm: Node, sibling: Node): elm is HTMLAnchorElement => (
+    elm === sibling.parentNode ||
+    sibling.parentNode && elm === sibling.parentNode.nextSibling
+  );
+  const isValidNode = (node: Node) => !isShortEnded(node) && isAnchor(node);
+
+  const rng = editor.selection.getRng();
+  const start = rng.startContainer;
+  const end = rng.endContainer;
+
+  const endOffsetChild = end.childNodes[rng.endOffset];
```

Comment:
So im guessing this will check childNodes on text nodes as well but that would produce undefined or null then the isBR would pick that up with it's truthy check above. Maybe it would be better to have do the element check first or maybe use options for all this instead of null checking.

Reply:
Added some comments on hot to make this a bit more safe.